server/src/main/java/org/opensearch/search/aggregations/AggregationCollectorManager.java

```diff
@@ -12,23 +12,24 @@
 import org.apache.lucene.search.CollectorManager;
 import org.opensearch.common.CheckedFunction;
 import org.opensearch.search.internal.SearchContext;
-import org.opensearch.search.profile.query.InternalProfileCollector;
 import org.opensearch.search.query.ReduceableSearchResult;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Collections;
 import java.util.List;
 
 /**
  * Common {@link CollectorManager} used by both concurrent and non-concurrent aggregation path and also for global and non-global
  * aggregation operators
+ *
+ * @opensearch.internal
  */
-class AggregationCollectorManager implements CollectorManager<Collector, ReduceableSearchResult> {
+public abstract class AggregationCollectorManager implements CollectorManager<Collector, ReduceableSearchResult> {
     protected final SearchContext context;
     private final CheckedFunction<SearchContext, List<Aggregator>, IOException> aggProvider;
     private final String collectorReason;
+    protected String collectorName;
```

Comment:
The `collectorName` is not needed here (is it not used), the Global / NonGlobal could have `private final String collectorName;` instead.

Reply:
Hi @reta, it has been addressed now. Please help to review it again, thanks.

Reply:
Thanks @ticheng-aws , LGTM !